plan/logical_plans.go

```diff
@@ -345,22 +345,29 @@ func (ds *DataSource) deriveTablePathStats(path *accessPath) error {
 	}
 	if pkCol == nil {
 		path.ranges = ranger.FullIntRange(false)
-		return nil
+		return false, nil
 	}
 	path.ranges = ranger.FullIntRange(mysql.HasUnsignedFlag(pkCol.RetType.Flag))
 	if len(ds.pushedDownConds) == 0 {
-		return nil
+		return false, nil
 	}
 	path.accessConds, path.tableFilters = ranger.DetachCondsForTableRange(ds.ctx, ds.pushedDownConds, pkCol)
 	path.ranges, err = ranger.BuildTableRange(path.accessConds, sc, pkCol.RetType)
 	if err != nil {
-		return errors.Trace(err)
+		return false, errors.Trace(err)
 	}
 	path.countAfterAccess, err = ds.statisticTable.GetRowCountByIntColumnRanges(sc, pkCol.ID, path.ranges)
-	return errors.Trace(err)
+	noIntervalRange := true
```

Comment:
Add comments here.

Reply:
how about s/noIntervalRange/allPointRanges/? This variable renaming can make code logic more clear.

Reply:
Agree